homeassistant/components/homekit_controller/__init__.py

```diff
@@ -227,6 +227,12 @@ async def async_setup_entry(hass: HomeAssistant, entry: ConfigEntry) -> bool:
 
     if not await conn.async_setup():
         del hass.data[KNOWN_DEVICES][conn.unique_id]
+        if (connection := getattr(conn.pairing, "connection")) and hasattr(
+            connection, "host"
+        ):
+            raise ConfigEntryNotReady(
+                f"Cannot connect to {connection.host}:{connection.port}"
+            )
```

Comment:
While right now its always going to be there since we only support IP, but since we are adding BLE/thread, longer term, it might be better to have pairing return a better error about why it cannot connect.

Reply:
Agree about returning a better error. I wonder if we should do that now. Either way I think I'd rather add something to aiohomekit rather than poking the connection object (and should probably add a _ prefix to that variable!)

Reply:
I'll open a PR to aiohomekit and clean this up after we get dev merged as I think its going to create conflicts now.

Reply:
Fixed in https://github.com/home-assistant/core/pull/75276